pkg/experimental/tests/rpm/source_date_epoch/BUILD

```diff
@@ -0,0 +1,133 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load(
+    "@rules_pkg//:mappings.bzl",
+    "pkg_filegroup",
+    "pkg_files",
+)
+load("@rules_pkg//experimental:rpm.bzl", "pkg_rpm")
+load("@rules_python//python:defs.bzl", "py_test")
+
+############################################################################
+# Test handling the source_date_epoch attribute
+############################################################################
+
+# The actual test.  It tests whether the contents match a particular desired
+# "manifest", most notably with regards to file structure.
+py_test(
+    name = "source_date_epoch_insource",
+    srcs = ["rpm_contents_vs_manifest_test.py"],
+    data = [":rpm_sde_insource_data"],
+    env = {"TEST_RPM": "rpm_sde_insource.rpm"},
+    main = "rpm_contents_vs_manifest_test.py",
+    tags = [
+        "no_windows",  # Windows doesn't have rpm(8) or rpmbuild(8)
+    ],
+    deps = [
+        "//experimental/tests/rpm:rpm_util",
+        "@rules_python//python/runfiles",
+    ],
+)
+
+# One cannot simply pass the output of pkg_rpm as runfiles content (#161).  This
+# seems to be the easiest way around this problem.
+sh_library(
+    name = "rpm_sde_insource_data",
+    testonly = True,
+    srcs = [":rpm_sde_insource"],
+)
+
+# The RPM (target under test)
+pkg_rpm(
+    name = "rpm_sde_insource",
+    srcs = [
+        ":pfg",
+    ],
+    architecture = "noarch",
+    description = """pkg_rpm test rpm description""",
+    license = "Apache 2.0",
+    release = "2222",
+    # Tue Mar 23 00:00:00 EDT 2021
+    source_date_epoch = 1616472000,
+    spec_template = "//experimental/tests/rpm:template-test.spec.in",
+    summary = "pkg_rpm test rpm summary",
+    version = "1.1.1",
+)
+
+pkg_filegroup(
+    name = "pfg",
+    srcs = [":pf"],
+)
+
+pkg_files(
+    name = "pf",
+    srcs = [":files"],
+    prefix = "test_dir",
+)
+
+genrule(
+    name = "files",
+    outs = [
+        "a",
+        "b",
+    ],
+    cmd = """
+    touch $(OUTS)
+    """,
+)
+
+############################################################################
+# Test handling the source_date_epoch attribute
+############################################################################
+
+py_test(
+    name = "source_date_epoch_from_file",
+    srcs = ["rpm_contents_vs_manifest_test.py"],
+    data = [":rpm_sde_fromfile_data"],
+    env = {"TEST_RPM": "rpm_sde_fromfile.rpm"},
+    main = "rpm_contents_vs_manifest_test.py",
+    tags = [
```

Comment:
As I was hinting at in the other comment, maybe an exec_compatible_with might be the thing to use here.
But... also happy to come back to it. That may force a minimum Bazel version and thus be a breaking change.

Reply:
As stated above, I think we should come back to this.  There are some other places where this would be useful in rules_pkg.

And WRT compatibility, we're already requring 4.0 as of #290, which uses the `json` module.  #319 does too.  The README states this as well.